low-level/sensors/impl/generic/node_sas_port.py

```diff
@@ -285,20 +290,30 @@ def check_and_send_alert(self):
         # Update sas ports status
         self.update_sas_ports_status()
 
-        if self.sas_phy_stored_alert == None:
-            # No info is stored for this node in Consul
+        # Check the version of stored alert
+        version = None
+        try:
+            # Try to get the version
+            # Exception will be raised if stored alert is None or no Version is available
+            version = self.sas_phy_stored_alert['version']
+        except:
+            logger.warning(f"Found no data or old data format for SASPortSensor, \
+                            updating data format to version {self.CURRENT_DATA_VERSION}")
+            # Versioning is not implemented or there is no data, write new data
             # Initialize dummy fault_resolved for all sas ports and conn
             self.sas_phy_stored_alert = {}
+            self.sas_phy_stored_alert['version'] = self.CURRENT_DATA_VERSION
             self.sas_phy_stored_alert['conn'] = 'fault_resolved'
             for i in range(0,self.NUM_SAS_PORTS):
                 self.sas_phy_stored_alert[i] = 'fault_resolved'
             # Save data to store
             store.put(self.sas_phy_stored_alert, self.SAS_PORT_SENSOR_DATA)
-        else:
+
+        if version == self.CURRENT_DATA_VERSION:
```

Comment:
this handling can be moved to a separate function. this will help with modularity when in future we need to deal with more versions.